### ISSUE TITLE
16 Review footer as solution to Main navigation broken at some screen widths

### ISSUE THREAD

**Issue by AlOneill** [FreeUKGen/MyopicVicar#1004]
Priority  16 (0	4	8	4)

At screen widths less than about 1100px, the social links are no longer visible -- not a big deal perhaps.

At screen widths between about 800 and 870px, **the "Help" link is not visible** -- this *is* a big deal, especially for a user on a device where the browser window cannot be resized.

Although the invisible links can be tabbed to, and will re-appear on-screen when in focus, this is unsatisfactory (and would remain so even after we make focus visually obvious) as it requires a fair degree of technical nous (or good luck!) to discover.

**Comment by AlOneill** [comment]
The basic issue is that we are trying to fit the proverbial quart into a pint pot. In other words, we have too many links in the main navigation, as it is currently set up.

The **Search page is the worst** because the "Transcriptions" link is longer than the "Search" link that replaces it on other pages.

There are a two obvious approaches: 

* reduce the number/character width of the links
* adjust the main nav break points that control the text size and spacing of the menu items

(A third could be to use drop-down menu items, combining some current items: this would need to be pure CSS to be accessible and would be tricky to set up to be nicely responsive -- could turn out to be impractical. A fourth could be to allow the main nav to wrap to a second line -- tricky, may be impractical.)

**Comment by Sherlock21** [comment]
If you  opt for a solution that ends up generating wrap-around, then please don't let the SEARCH button go off the visible screen page bottom.

**Comment by Captainkirkdawson** [comment]
Is this a regression  we did with the accessibility adjustments or was it always there

**Comment by PatReynolds** [comment]
I  suggest we remove 'donate' from the menu, and have it as a separate item (button), either just underneath the title or just above the footer.

**Comment by AlOneill** [comment]
Kirk, the issue has been present for a while.  It has not been affected by any of the accessibility changes. 

The increase in the maximum width of a 'container' has allowed us to increase the number of main menu items, but at the expense of how much of that menu remains on-screen at less than maximum screen/window widths, especially on the Search page.

Pat, I am not clear where you mean by 'just underneath the title'. Putting a button near the top of a page will have an impact on the 'clutter' (if I may call it that!) at the top of a page, forcing the main content down further. On the face of it, I can't see how this could be done nicely.

There is a (small) link to 'Donate' in the footer already -- perhaps we need to re-think how the footer contents are presented, so that links to Donate and Volunteer are more prominent there. Neither of these are links that most visitors will use on a regular basis: so they probably do not belong in the main menu.

**Comment by PatReynolds** [comment]
Yes - review footer.

**Comment by Captainkirkdawson** [comment]
Need to respond to 24 management on charity status changes so should deal with other issues with footer at same time

**Comment by AlOneill** [comment]
The proposed new footer requires a number of markup and CSS changes. The image shows the revised footer (as discussed by Pat, Rich and Alison), plus a number of tentative text changes.

Because the changes are extensive, I see implementing a revised footer as part of the "last piece of the Accessibility jigsaw" -- updating application.html.erb, which provides the 'frame' for all pages. And because the other pieces of the jigsaw are not all in place, a *full* update would not be appropriate at this point.

![footer-rev1 0](https://cloud.githubusercontent.com/assets/14852554/21463260/79668800-c95e-11e6-81f1-88d66020c09e.png)

**Comment by Captainkirkdawson** [comment]
Not sure why Social links has the 2 links and then there are 2 links under project by ukgen. All 4 links refer to ukgen links all be it 2 are hosted internally and 2 externally . Would it not be better to have the 2 columns under a simple title of  FreeUKGenealogy. Put the freeukGenealogy icon up with the other 2 icons

**Comment by PatReynolds** [comment]
I'm  not sure what probem this is trying to solve.

The Free UK Genealogy icon is also giving the information that FreeREG is one of our projects.  So maybe it needs to go at the top.
Then 'our other projects, helping you to find your family history'
then the two logos
I wouldn't say 'Policies' are a 'Social Link', but yes, Blog (probably to be rechristened 'News') could go there.

**Comment by Captainkirkdawson** [comment]
My comments related to the proposed new footer appearing unbalanced at least in my mind
![balance](https://cloud.githubusercontent.com/assets/2874694/21503749/d3191cc2-cc16-11e6-8498-2d7079a78a4b.JPG)

**Comment by Captainkirkdawson** [comment]
I agree much of this is personal preference. To my eye having 2 links under the freeukgen icon is disturbing.
 3 columns of links and then the icon would be more balanced for me

**Comment by PatReynolds** [comment]
Icon headed column in the middle?

Pat Reynolds
Executive Director, Free UK Genealogy

**Comment by AlOneill** [comment]
Thanks for the comments -- two alternatives for your consideration ...

**Version 2**

![footer-rev1 2](https://cloud.githubusercontent.com/assets/14852554/21547914/f7b0d4aa-cde0-11e6-9d22-c76f10ae072b.png)

**Version 3**

![footer-rev1 3](https://cloud.githubusercontent.com/assets/14852554/21547926/082aea64-cde1-11e6-964d-6978840a469e.png)

**Comment by edickens** [comment]
I prefer Version 3.  E
    
      On 29/12/2016 16:16, AlOneill wrote:
    
    
      Thanks for the comments -- two alternatives for your
        consideration ...
      Version 2
      
      Version 3
      
      
      
      
        
          
          
        
        
      
      {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/FreeUKGen/MyopicVicar","title":"FreeUKGen/MyopicVicar","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/FreeUKGen/MyopicVicar"}},"updates":{"snippets":[{"icon":"PERSON","message":"@AlOneill in #1004: Thanks for the comments -- two alternatives for your consideration ...\r\n\r\n**Version 2**\r\n\r\n![footer-rev1 2](https://cloud.githubusercontent.com/assets/14852554/21547914/f7b0d4aa-cde0-11e6-9d22-c76f10ae072b.png)\r\n\r\n**Version 3**\r\n\r\n![footer-rev1 3](https://cloud.githubusercontent.com/assets/14852554/21547926/082aea64-cde1-11e6-964d-6978840a469e.png)\r\n"}],"action":{"name":"View Issue","url":"https://github.com/FreeUKGen/MyopicVicar/issues/1004#issuecomment-269653095"}}}

**Comment by Captainkirkdawson** [comment]
Personally I would like to see a variation of version 3 that had 4
"columns" in the footer; the 3 text ones and 1 occupied by the freeukgen
icon

Kirk Dawson
5220 Riverside Drive
Fairmont Hot Springs, B.C.
V0B 1L1

**Comment by PatReynolds** [comment]
include donate button

**Comment by Captainkirkdawson** [comment]
@AlOneill  are we ready to test this on test3?

**Comment by AlOneill** [comment]
In short, no. 
Do not know what "donate button" is intended to look like: our standard button or something else? 
Have not begun changes to erb file.

**Comment by Captainkirkdawson** [comment]
The donate is to go in the search results page. I was referring to completion of the work  as of 3 weeks ago. Understand the erb changes are to come

**Comment by richpomfret** [comment]
David to create donate button. Review still in-progress.

**Comment by PatReynolds** [comment]
@AlOneill to consider possible changes to Free UK Genealogy logo for use against this background.

**Comment by richpomfret** [comment]
Let's look at creating a mock-up for this (considering the other story relating to site stats).

**Comment by richpomfret** [comment]
Stats are STAYING. Alison to take a look/review.

**Comment by PatReynolds** [comment]
Footer also needs to be able to handle 'bouquets' in due course (see #1195).  Sorry @AlOneill !

**Comment by Captainkirkdawson** [comment]
@AlOneill  is this still open and a requirement

**Comment by PatReynolds** [comment]
I believe this is resolved on systemwide, so moved to Needs Review
